File: qqbot-protocol/src/main/java/com/mieai/qqbot/protocol/error/QqApiError.java
package com.mieai.qqbot.protocol.error;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonProperty;

/** Error payload returned by QQ OpenAPI endpoints. */
@JsonIgnoreProperties(ignoreUnknown = true)
public record QqApiError(
        @JsonAlias("err_code") int code,
        String message,
        @JsonProperty("trace_id") String traceId) {}

File: qqbot-protocol/src/main/kotlin/com/mieai/qqbot/protocol/error/QqApiError.kt
package com.mieai.qqbot.protocol.error

import com.fasterxml.jackson.annotation.JsonAlias
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

/** Error payload returned by QQ OpenAPI endpoints. */
@JsonIgnoreProperties(ignoreUnknown = true)
data class QqApiError(
    @JsonAlias("err_code") val code: Int,
    val message: String?,
    @JsonProperty("trace_id") val traceId: String?,
)
